src/main/java/com/dev/cinema/dao/impl/GenericDaoImpl.java

```diff
@@ -9,12 +9,10 @@
 import org.hibernate.SessionFactory;
 import org.hibernate.Transaction;
 import org.hibernate.query.Query;
-import org.springframework.stereotype.Repository;
 
-@Repository
 public abstract class GenericDaoImpl<T> implements GenericDao<T> {
     public static final Logger logger = Logger.getLogger(GenericDaoImpl.class);
-    SessionFactory sessionFactory;
+    public SessionFactory sessionFactory;
```

Comment:
we can use protected here and it's recommended to use as strict access modifier as possible